src/raw/unix.rs

```diff
@@ -337,7 +359,54 @@ impl Socket {
         }
     }
 
-    ///Accepts incoming connection.
+    ///Accept a new incoming client connection and return its files descriptor and address.
+    ///
+    ///By default the newly created socket will be inheritable by child processes and created
+    ///in blocking I/O mode. This behaviour can be customized using the `flags` parameter:
+    ///
+    /// * `AcceptFlags::NON_BLOCKING`    – Mark the newly created socket as non-blocking
+    /// * `AcceptFlags::NON_INHERITABLE` – Mark the newly created socket as not inheritable by client processes
+    ///
+    ///Depending on the operating system's availablility of the `accept4(2)` system call this call
+    ///either pass the flags on to the operating system or emulate the call using `accept(2)`.
+    pub fn accept4(&self, flags: c_int) -> io::Result<(Socket, net::SocketAddr)> {

```

Comment:
Why not accept only `AcceptFlags` ? This way you can control that user will not try to pass some bad flag